SCRUM-42-fe-navbar

AidaBur opened the pull request:
1. Added a new Navbar component
2. Displays on all routes, adapting to auth state and current page
3. Shows anchor links + Login/Signup on the landing page before login
4. Displays "+ Add post" and avatar with dropdown after login
5. Adds a hover-based dropdown menu with:
- My items (/app/profile?tab=items)
- Account info (/app/profile?tab=info)
- Logout
<img width="1792" alt="Screen Shot 2025-04-21 at 2 16 05 PM" src="https://github.com/user-attachments/assets/89278630-b256-43f9-a736-a7f79f2e1390" />
<img width="1792" alt="Screen Shot 2025-04-21 at 2 15 45 PM" src="https://github.com/user-attachments/assets/08c62fce-69d3-4b70-a2d4-e90662067f56" />
<img width="1792" alt="Screen Shot 2025-04-21 at 2 15 32 PM" src="https://github.com/user-attachments/assets/e8f85fe2-ecce-48be-a511-c9f1e717af8d" />

MariaZasypkina reviewed (approved):
Looking great for me)

mateuspiresl commented on src/components/Navbar.jsx:
Seems that we have three states here:

- Landing page + user logged out
- Landing page + user logged in
- Auth page + user logged out
- App page + user logged in

But we should not allow the user to access the landing and auth pages when the user is logged in.

So we should have:

- Landing page + user logged out
- Auth page + user logged out
- App page + user logged in

Whenever the logged in user reaches the landing or auth pages, redirect to the app page.

Whenever the logged out user reaches the app pages, redirect to the login page.

mateuspiresl commented on src/components/Navbar.jsx:
```diff
@@ -0,0 +1,194 @@
+  return (
+    <nav className="w-full bg-white z-50 relative">
+      <div className="max-w-[1440px] px-[100px] mx-auto py-4 min-h-[100px] flex items-center justify-between">
+        <Link to="/" className="flex items-center gap-2">
```
The logo should link to the correct page depending on whether the user is logged in or out.

AidaBur commented:
This PR updates the Navbar and routing logic to handle redirects for authenticated and unauthenticated users.

- Replaced useEffect with direct condition for `isLanding` to reduce unnecessary renders.
- Updated `<Link>` on the logo to redirect to `/app/posts` when user is logged in.
- Added `RedirectLogic` wrapper in the router to:
  - Redirect logged-in users away from landing/auth pages to the app.
  - Redirect logged-out users from `/app` routes to `/login`.
